src/middleware/middleware.js

```diff
@@ -24,11 +29,12 @@ const validateUser = (req, res, next) => {
         return res.status(500).send('Internal Server Error')
       }
 
-      if (result && (result.cna_short_name !== cna || uuidAPIKey.toAPIKey(result.secret) !== secret)) {
-        logger.warn('User authentication FAILED for ' + submitter)
-        return res.status(500).json({ message: 'User Authentication FAILED' })
-      }
-
+      if (result) {
+        if ((result.cna_short_name !== cna) || (uuidAPIKey.toAPIKey(result.secret) !== secret)) {
+          logger.warn('User authentication FAILED for ' + submitter)
+          return res.status(401).json({ message: 'User Authentication FAILED.' })
+        }
+      } 
       logger.info('SUCCESSFUL user authentication for ' + submitter)
       next()
```

Comment:
Bug: If the user is not found in the database it will pass the user validation.
Include an if statement that if no user is found, the user validation fails immediately.